#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: bash(1): Document ~/.bash_login and ~/.profile in FILES section

--- bash/doc/bash.1~	2004-04-14 07:57:10.000000000 +0200
+++ bash/doc/bash.1	2004-04-27 10:24:10.000000000 +0200
@@ -8286,6 +8286,18 @@
 .FN ~/.bash_profile
 The personal initialization file, executed for login shells
 .TP
+.FN ~/.bash_login
+Alternative personal initialization file, executed for login shells, if
+.FN ~/.bash_profile
+is not present.
+.TP
+.FN ~/.profile
+Alternative personal initialization file, executed for login shells, if both
+.FN ~/.bash_profile
+and
+.FN ~/.bash_login
+are not present.
+.TP
 .FN ~/.bashrc
 The individual per-interactive-shell startup file
 .TP
